addons/mail/__manifest__.py

```diff
@@ -51,222 +51,25 @@
     'application': True,
     'assets': {
         'web._assets_primary_variables': [
-            'mail/static/src/scss/variables.scss',
+            'mail/static/src/scss/variables/primary_variables.scss',
         ],
         'web.assets_backend': [
-            'mail/static/src/js/core/translation.js',
-            'mail/static/src/js/many2many_tags_email.js',
-            'mail/static/src/js/m2x_avatar_user.js',
-            'mail/static/src/js/field_char.js',
-            'mail/static/src/js/document_viewer.js',
-            'mail/static/src/js/basic_view.js',
-            'mail/static/src/js/systray/systray_activity_menu.js',
-            'mail/static/src/js/tours/mail.js',
-            'mail/static/src/js/tools/debug_manager.js',
-            'mail/static/src/js/custom_filter_item.js',
-            'mail/static/src/js/utils.js',
-            'mail/static/src/js/activity.js',
-            'mail/static/src/js/views/activity/activity_view.js',
-            'mail/static/src/js/views/activity/activity_model.js',
-            'mail/static/src/js/views/activity/activity_controller.js',
-            'mail/static/src/js/views/activity/activity_renderer.js',
-            'mail/static/src/js/views/activity/activity_record.js',
-            'mail/static/src/js/views/activity/activity_cell.js',
-            'mail/static/src/js/emojis.js',
-            'mail/static/src/js/emojis_mixin.js',
-            'mail/static/src/js/field_emojis_common.js',
-            'mail/static/src/js/field_char_emojis.js',
-            'mail/static/src/js/field_text_emojis.js',
-            'mail/static/src/scss/emojis.scss',
+            'mail/static/src/js/**/*.js',
+            ('remove', 'mail/static/src/js/main.js'), # main is prod only
+            'mail/static/src/utils/*/*.js',
+            'mail/static/src/utils/*/*.scss',
+            'mail/static/src/utils/utils.js',
             'mail/static/src/variables.scss',
```

Comment:
Shouldn't this variable file be merged into the other one?

Reply:
Seems reasonable, decided to keep it this way at first to make sure nothing was broken, but I'll move those definitions to `primary_variables.scss`

Reply:
@seb-odoo after trying it out, it looks like I can't merge them. Some definitions needs to be in primary_variables because other modules depend on them to be there (eg im_livechat has some files in assets_frontend that use some of these variable definitions), and one variable definition depends on bootstrap variables being defined but they're only loaded later. I've reorganized them in a way that makes sense to me, let me know if you'd rather have them organized differently.

Reply:
Sounds good, just I would not move notification list scss to utils, it moves it further away from the feature where it is used which I think is going the wrong way to me.

Reply:
Not sure where else it makes sense to put them, they're shared functionality between multiple component and as such they strike me as being "utils". Would you put it at the root of the components folder?

Reply:
Just keep it where it was, in the notification list component.
Ideally we shouldn't have a split by component but by feature anyway, so it would be on the corresponding feature folder. The closest we have to a feature folder now is the component folder itself.